lib/middleware.ts

```diff
@@ -113,6 +113,97 @@ var defaultHandlers = [
   () => ({title: "request-finished"}),
 ];
 
+const defaultContextHandlers = [
+  (req) => ({"request-id": req.headers["x-request-id"]}),
+  (req) => ({"session-id": req.sessionID}),
```

Comment:
Since most of our services don't have session-id, I would make this an optional handler.